### ISSUE TITLE
Migrate to null safety

### ISSUE THREAD

**Issue by lem8r** [joanpablo/reactive_forms#86]
Hi!

Thanks for sharing your package.
Please consider migrating it to null safety.
All dependencies already has null-safety version so it's not blocking factor.

**Comment by joanpablo** [comment]
Hi @lem8r,

Thanks for the advice. I will move it to null-safety.

**Comment by radvansky-tomas** [comment]
+1

**Comment by gmcdowell** [comment]
+1

**Comment by gmcdowell** [comment]
Hi,

I have a [working version](https://github.com/tierratechnologies/reactive_forms) of null-safety changes that I am just doing some QA on.
Hope to have a PR in soon.

**Comment by joanpablo** [comment]
Hi guys, 

Thanks for this issue, I see is very useful. I will need a little more time, I have a release soon that needs some adjustments first. And after that, I will work on this issue.

**Comment by derolf** [comment]
> Hi,
> 
> I have a [working version](https://github.com/tierratechnologies/reactive_forms) of null-safety changes that I am just doing some QA on.
> Hope to have a PR in soon.

Bugs:

Line 855 models.dart should be:

```Dart
    if (_focusController != null) {
      unregisterFocusController(_focusController);
    }
```

**Comment by vcanuel** [comment]
HI, please note that intl 0.17.0 is required now (with flutter 2)

**Comment by joanpablo** [comment]
Hi @vcanuel,

I have just released 9.1.0. This version includes the latest version of intl: 0.17.0.

Reactive Forms is not yet null safety but at least it includes the intl latest version.

Thanks and Best Regards

**Comment by kuhnroyal** [comment]
@joanpablo  I want to reiterate my points about type-safety before null-safety that I made in different places across the project.

`dynamic` can be `Object` or `Object?` and this is a huge problem for the null-safety migration tool, as well as for manual migration. Currently a lot of types in the library are or return `Type<dynamic>`.
In my opinion this needs to be fixed before there is a chance for a null-safety version that is actually useful to someone and doesn't just return `dynamic` everywhere.

I suggest these steps:

1. Remove current form builder
2. Enable this in `analysis_options`
```yaml
analyzer:
  language:
    strict-inference: true
    strict-raw-types: true
  strong-mode:
    implicit-dynamic: false
    implicit-casts: false
```
3. Fix all type errors by adding generic types and returning the correct ones
4. Create a new for builder that works with the new generic types
5. Migrate to null-safety and release `10.0.0`

**Comment by joanpablo** [comment]
Hi @kuhnroyal,

Thanks for your great warning, I will take it into account when rethinking the dynamic type system. Right now I'm doing the migration manually and analyzing each line of code in the project. I have configured the project to be fully null-safety compatible, and now I'm fixing incompatibilities line by line. This is a very good exercise to rethink each decision I have made in the project. After the project compiles and all test runs successfully I will change to the more restrictive configurations in "analysis_options" and again rethink all patterns about the data types. I think you are right about dynamicity and I totally agree with you, that there are some changes that need to be done.

Thanks

**Comment by radvansky-tomas** [comment]
That dynamic part is probably from whole reactive forms JS/TS vision, I get that and agree that we should change it and adopt more type safe code, with generics for places where it make sense.

**Comment by prakash-indorkar** [comment]
Hi guys!, thanks for the great contribution. Just wondering how far are we for null safety release. I'm new to this package, I've been using flutter_form_builder before but this package seems much more advanced and featureful. I have already moved my app to null safety except for the forms. Just waiting for the null safe version of this package to use it properly. Thanks in advance.

**Comment by cedvdb** [comment]
@joanpablo I hope you don't go too far down the refactor rabbit hole. If you feel like it's necessary, so be it though.

**Comment by prakash-indorkar** [comment]
I agree with @cedvdb. I think at least pre-release should be released.

**Comment by joanpablo** [comment]
Hi,

I have released version 10.0.0 with null safety. Please test it, and open new issues if you have some questions. :)
